New Terms: C/C++ `strtok()` (#871)
Codecademy/docs: content/c/concepts/strings/terms/strtok/strtok.md

---
Title: 'strtok()'
Description: 'Breaks a string into a series of tokens using a list of delimiters.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Data Types'
  - 'Characters'
  - 'Arrays'
  - 'Functions'
CatalogContent:
  - 'learn-c'
  - 'paths/computer-science'
---

The **`strtok()`** function breaks a string into a series of tokens using a list of delimiters.

## Syntax

```pseudo

> char *token = strtok(string, delimiters);

Reviewer: ```suggestion
char *token = strtok(string, delimiters);

token = strtok(NULL, delimiters);
```

I feel like adding this line better supports the points made in the paragraph  below.

```

> The `strtok()` function splits the string `string` into tokens on occurrences of any character in the string `delimiters` and returns the pointer to the first token. Subsequent calls to strtok with `string` = `NULL` will return subsequent tokens. It returns a `NULL` pointer when there are no more tokens to retrieve.

Reviewer: ```suggestion
The `strktok()` function splits the `string` parameter into tokens based on one or more `delimiters`, and returns the [pointer](https://www.codecademy.com/resources/docs/c/pointers) to the first token. Subsequent calls to `strtok()`, with `string` set to `NULL`, return a pointer to the next tokenized string. When there are no tokens left to retrieve, a `NULL` pointer is returned.
```

## Example

The following example splits up a comma-delimited list and prints out the result:

```c
#include <string.h>
#include <stdio.h>

int main () {
   char str[] = "Comma,Delimited,List";
   const char s[] = ",";
   char *token = strtok(str, s);

   while( token != NULL ) {
      printf( "%s\n", token );
      token = strtok(NULL, s);
   }
}
```

> The above example results in the following output:

Reviewer: ```suggestion
This results in the following output:
```

---
Title: 'strtok()'
Description: 'Breaks a string into a series of tokens using a list of delimiters.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Data Types'
  - 'Characters'
  - 'Arrays'
  - 'Functions'
CatalogContent:
  - 'learn-c'
  - 'paths/computer-science'
---

The **`strtok()`** function breaks a string into a series of tokens using a list of delimiters.

## Syntax

```pseudo
char *token = strtok(string, delimiters);

token = strtok(NULL, delimiters);
```

> The `strktok()` function splits the `string` parameter into tokens based on one or more `delimiters`, and returns the [pointer](https://www.codecademy.com/resources/docs/c/pointers) to the first token. Subsequent calls to `strtok()`, with `string` set to `NULL`, return a pointer to the next tokenized string. When there are no tokens left to retrieve, a `NULL` pointer is returned.

Reviewer 2: ```suggestion
The `strktok()` function splits the `string` parameter into tokens based on one or more `delimiters`, and returns the [pointer](https://www.codecademy.com/resources/docs/c/pointers) to the first token. 

Subsequent calls to `strtok()`, with `string` set to `NULL`, return a pointer to the next tokenized string. When there are no tokens left to retrieve, a `NULL` pointer is returned.
```

---
Title: 'strtok()'
Description: 'Breaks a string into a series of tokens using a list of delimiters.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Data Types'
  - 'Characters'
  - 'Arrays'
  - 'Functions'
CatalogContent:
  - 'learn-c'
  - 'paths/computer-science'
---

The **`strtok()`** function breaks a string into a series of tokens using a list of delimiters.

## Syntax

```pseudo
char *token = strtok(string, delimiters);

token = strtok(NULL, delimiters);
```

> The `strktok()` function splits the `string` parameter into tokens based on one or more `delimiters`, and returns the [pointer](https://www.codecademy.com/resources/docs/c/pointers) to the first token.

Reviewer 3: ```suggestion
The `strtok()` function splits the `string` parameter into tokens based on one or more `delimiters`, and returns the [pointer](https://www.codecademy.com/resources/docs/c/pointers) to the first token.
```
